java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/db/MDEntityClassFromDB.java

```diff
@@ -0,0 +1,144 @@
+                if (project == null) {
+                    client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoProject(folder.getPath())));
+                    return null;
+                }
+                SourceGroup sourceGroup = getFolderSourceGroup(ProjectUtils.getSources(project).getSourceGroups(JavaProjectConstants.SOURCES_TYPE_JAVA), folder);
+                if (sourceGroup == null) {
+                    client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoSourceGroup(folder.getPath())));
+                    return null;
+                }
+                DatabaseConnection connection = ConnectionManager.getDefault().getPreferredConnection(true);
+                if (connection == null) {
+                    client.showMessage(new MessageParams(MessageType.Error, Bundle.MSG_NoDbConn()));
+                    return null;
+                }
+                ConnectionManager.getDefault().connect(connection);
+                Connection conn = connection.getJDBCConnection();
+                ResultSet rs = conn.getMetaData().getTables(conn.getCatalog(), conn.getSchema(), "%", new String[]{"TABLE", "VIEW"}); //NOI18N
+                List<QuickPickItem> dbItems = new ArrayList<>();
+                while (rs.next()) {
+                    dbItems.add(new QuickPickItem(rs.getString("TABLE_NAME"))); //NOI18N
+                }
+                return client.showQuickPick(new ShowQuickPickParams(Bundle.MSG_SelectTables(), true, dbItems))
+                        .thenApply(items -> items.stream().map(item -> item.getLabel()).collect(Collectors.toList()))
+                        .thenApply(tables -> {
+                            EntitiesFromDBGenerator generator = new EntitiesFromDBGenerator(tables, false, false, false,
+                                    EntityRelation.FetchType.DEFAULT, EntityRelation.CollectionType.COLLECTION,
+                                    getPackageForFolder(sourceGroup, folder), sourceGroup, connection, project, null, createPersistenceGenerator(TYPE_MICRONAUT));
+                            ProgressContributor pc = BasicAggregateProgressFactory.createProgressContributor("entity"); //NOI18N\
+                            try {
+                                return generator.generate(pc);
+                            } catch (IOException | SQLException ex) {
+                                client.showMessage(new MessageParams(MessageType.Error, ex.getMessage()));
+                            }
+                            return null;
+                        })
+                        .thenApply(generated -> {
+                            if (generated != null) {
+                                return generated.stream().map(fo -> Utils.toUri(fo)).collect(Collectors.toList());
+                            }
+                            return null;
+                        });
+            } catch (Exception ex) {
```

Comment:
Report a message to the client - may be the DB connection failed ?

Reply:
Code moved to `MicronautEntity.lspCreate()` and message to client added.